Cython/Compiler/ParseTreeTransforms.py

```diff
@@ -2560,6 +2562,19 @@ def visit_DefNode(self, node):
             node = node.as_cfunction(scope=self.scope)
         return node
 
+    def visit_CClassDefNode(self, node, pxd_def=None):
+        if pxd_def is None:
+            pxd_def = self.scope.lookup(node.class_name)
+        if pxd_def:
+            if not pxd_def.defined_in_pxd:
+                return node
```

Comment:
Is this really relevant? If I read it correctly, it excludes cdef classes that are not defined in a .pxd file. That shouldn't matter for `auto_cpdef`, should it?

Reply:
I have checked that and compiling following file `cython/Cython/Plex/DFA.py` fails when it is removed:
```
$ python cython.py -3 /home/matus/dev/cython/Cython/Plex/DFA.py

Error compiling Cython file:
------------------------------------------------------------
...
    """
    Helper class used by nfa_to_dfa() to map back and forth between
    sets of states from the old machine and states of the new machine.
    """

    def __init__(self, new_machine):
   ^
------------------------------------------------------------

Cython/Plex/DFA.py:100:4: Special methods must be declared with 'def', not 'cdef'

Error compiling Cython file:
------------------------------------------------------------
...
    """
    Helper class used by nfa_to_dfa() to map back and forth between
    sets of states from the old machine and states of the new machine.
    """

    def __init__(self, new_machine):
   ^
------------------------------------------------------------

Cython/Plex/DFA.py:100:4: C method '__init__' not previously declared in definition part of extension type 'StateMap'

Error compiling Cython file:
------------------------------------------------------------
...
            if priority > best_priority:
                best_action = state.action
                best_priority = priority
        return best_action

    def new_to_old(self, new_state):
   ^
------------------------------------------------------------

Cython/Plex/DFA.py:132:4: C method 'new_to_old' not previously declared in definition part of extension type 'StateMap'

Error compiling Cython file:
------------------------------------------------------------
...
        Convert a set of states into a uniquified
        sorted tuple suitable for use as a dictionary key.
        """
        return tuple(sorted(state_set))

    def dump(self, file):
   ^
------------------------------------------------------------

Cython/Plex/DFA.py:143:4: C method 'dump' not previously declared in definition part of extension type 'StateMap'

Error compiling Cython file:
------------------------------------------------------------
...
    """
    Helper class used by nfa_to_dfa() to map back and forth between
    sets of states from the old machine and states of the new machine.
    """

    def __init__(self, new_machine):
   ^
------------------------------------------------------------

Cython/Plex/DFA.py:100:4: Return with value in void function
```